resolver/type_syntax.cc

```diff
@@ -463,6 +463,22 @@ ParsedSig parseSigWithSelfTypeParams(core::Context ctx, const ast::Send &sigSend
     return sig;
 }
 
+std::optional<std::string> autoCorrectOrOfOnlyClasses(core::Context ctx, core::TypePtr type) {
+    if (auto *o = core::cast_type<core::OrType>(type) ) {
+        auto left = autoCorrectOrOfOnlyClasses(ctx, o->left);
+        auto right = autoCorrectOrOfOnlyClasses(ctx, o->right);
```

Comment:
I think that this has the unfortunate side effect of converting this:

```ruby
T.class_of(T.any(A, B, C))
```

into this:

```ruby
T.any(T.class_of(A), T.any(T.class_of(B), T.class_of(C)))
```

when it would be nice if we could turn this directly into:

```ruby
T.any(T.class_of(A), T.class_of(B), T.class_of(C))
```

What do you think about writing it that way?

One way to do this would be to return a vector of either symbols or strings, and then using something like

```cpp
fmt::format("T.any({})", fmt::map_join(symbols, ", ", [&](auto sym) {
  return fmt::format("T.class_of({})", sym.show(gs);
}));
```

Reply:
That approach sounds reasonable to me, not much harder to implement.